frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/hooks/useMilestoneProgressionInfo.ts

```diff
@@ -0,0 +1,36 @@
+import { formatDistanceToNow, addMinutes } from 'date-fns';
+import { useLocationSettings } from 'hooks/useLocationSettings';
+import { formatDateYMDHM } from 'utils/formatDate.ts';
+import type { MilestoneStatus } from '../ReleasePlanMilestone/ReleasePlanMilestoneStatus.tsx';
+
+export const useMilestoneProgressionInfo = (
```

Comment:
I'd extract a fn without the hook and test it in unit tests as there's some involved logic in there